### PR TITLE
minor 'notes' bugfix, add 'loop' meta-command to shell

### PR DESCRIPTION
**Pull request by ixcat**

### CODE REVIEW

File: pipeline/shell.py [hunk 1]
```diff
@@ -519,6 +519,31 @@ def sync_and_external_cleanup():
...
+
+    log.info('running {} in a loop'.format(cmd))
+
+    nruns = nerrs = 0
+
+    while True:
+        log.info('{} loop round {} (nerrs: {})'.format(cmd, nruns, nerrs))
+        try:
+            cmd_fn(*args)
+        except Exception as e:
```
Comment (ttngu207) [inline pipeline/shell.py:536]: I'd suggest only catching connection-related errors only, and allow the looping. We probably still want the ingestion to crash for any other type of errors

Comment (ixcat) [reply pipeline/shell.py:536]: Agree - as I understand it currently, the datajoint:pymysql exception heirarchy isn't 100% clean which is why I did this.

Though, might make sense to use this as a testbed and just catch `dj.errors.LostConnectionError` - anything that falls through probably is something to fix in DataJoint.

will adjust and we'll see how it goes.

Comment (ixcat) [reply pipeline/shell.py:536]: Actually - making this connection error specific is better for our current use case (detecting only connection errors), but less general purpose (e.g. as a background job to run ingest or other tasks) - in the latter case, sometimes simply correcting input data fixes the 'loop' and wouldn't indicate a code problem, or other kinds of scenarios (e.g. s3 error types for sync report loops, etc).

Given that there is a human factor which would detect ingest errors (why don't I see my session? -> check logs) i'm somewhat in favor of leaving this generic, despite the potential drawbacks for some cases that you highlight.

will ping for conversation to sort it out.

Comment (ttngu207) [reply pipeline/shell.py:536]: Right, I see your point, users can take this as 2 modes of ingestion:
1. I want to make sure I'm aware of the errors, crash if there's any error -> don't use `loop`
2. I will let this run overnight, bypass all errors, I'll check log later -> use `loop`

**Review by ttngu207 — APPROVED** [review]

**Pull request merged**